src/Files.App/Helpers/DynamicDialogFactory.cs

```diff
@@ -66,18 +66,18 @@ public static DynamicDialog GetFor_RenameDialog()
 
 			TextBlock tipText = new()
 			{
-				Text = "RenameDialogSymbolsTip/Text".GetLocalizedResource(),
+				Text = "InvalidFilename/Text".GetLocalizedResource(),
 				Margin = new Microsoft.UI.Xaml.Thickness(0, 0, 4, 0),
 				TextWrapping = Microsoft.UI.Xaml.TextWrapping.Wrap,
 				Opacity = 0.0d
 			};
 
 			inputText.TextChanged += (textBox, args) =>
 			{
-				tipText.Opacity = FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) ? 1.0d : 0.0d;
-				dialog!.ViewModel.DynamicButtonsEnabled = string.IsNullOrWhiteSpace(inputText.Text) || FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) 
-														? DynamicDialogButtons.Cancel 
-														: DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
+				tipText.Opacity = FilesystemHelpers.IsValidForFilename(inputText.Text) ? 0.0d : 1.0d;
+				dialog!.ViewModel.DynamicButtonsEnabled = FilesystemHelpers.IsValidForFilename(inputText.Text)
+														? DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel
+														: DynamicDialogButtons.Cancel;
```

Comment:
We can win one test by stocking the value in a variable before hand and reusing them.
```suggestion
				var isInputValid = FilesystemHelpers.IsValidForFilename(inputText.Text);
				tipText.Opacity = isInputValid ? 0.0d : 1.0d;
				dialog!.ViewModel.DynamicButtonsEnabled = isInputValid 
														? DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel
														: DynamicDialogButtons.Cancel;
```